core/arch/arm/mm/core_mmu.c

```diff
@@ -460,13 +474,31 @@ static void dump_mmap_table(struct tee_mmap_region *memory_map)
 	}
 }
 
+static void sort_mmap_table_by_va(struct tee_mmap_region *memory_map)
```

Comment:
We're sorting at a few places in this file, please make this function generic by taking the comparison function as an argument. `add_phys_mem()` doesn't something special while populating the array so that one is excused.

While you're at it add a short comment why this particular sorting algorithm is used, I suppose it has a formal name also somewhere.